pkg/memory-monitor/src/monitor/procfs.c

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2024 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <linux/limits.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <syslog.h>
+#include <unistd.h>
+
+#include "config.h"
+#include "monitor.h"
+#include "util.h"
+
+#include "procfs.h"
+
+static unsigned long procfs_get_rss(int pid) {
+    char path[PATH_MAX];
+    FILE *fp;
+    unsigned long rss = 0;
+
+    // Create the path to the status file for the process
+    sprintf(path, "/proc/%d/status", pid);
+
+    // Open the file
+    fp = fopen(path, "r");
+    if (fp == NULL) {
+        syslog(LOG_INFO, "Failed to open status file for process %d\n", pid);
+        return 0;
+    }
+
+    char line[256];
+    while (fgets(line, sizeof(line), fp) != NULL) {
+        // If the line starts with "VmRSS", extract the value
+        if (strncmp(line, "VmRSS", strlen("VmRSS")) == 0) {
+            sscanf(line, "VmRSS: %lu kB", &rss);
+            break;
+        }
+    }
+
+    // Close the file
+    fclose(fp);
+
+    // Convert the value to bytes
+    unsigned long rss_bytes = rss << 10;
+    return rss_bytes;
+}
+
+int procfs_check_rss(int pid, unsigned long threshold) {
+    unsigned long rss;
+    static bool handler_executed = false;
+
+    // Get the RSS of the process
+    rss = procfs_get_rss(pid);
+    if (rss == 0) {
+        syslog(LOG_INFO, "Failed to get the RSS of the process %d\n", pid);
+        return 1;
+    }
+
+    // Check if the RSS exceeds the threshold
+    if (rss > threshold) {
+        // Run the handler script only per one threshold exceed
+        if (handler_executed)
+            // The handler script has already been executed
+            return 0;
+        syslog(LOG_INFO, "----- Zedbox threshold is reached -----\n");
+        char event_msg[256];
```

Comment:
```suggestion
        char event_msg[103];
```
At least that's what I count ... ;-)

Reply:
We can be generous and give them 153 bytes more) But I can replace it with some meaningful constant if you wish.